core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -168,7 +168,8 @@ private[spark] class Executor(
             case Some(existingReaper) => interruptThread && !existingReaper.interruptThread
           }
           if (shouldCreateReaper) {
-            val taskReaper = new TaskReaper(taskRunner, interruptThread = interruptThread)
+            val taskReaper = new TaskReaper(
+              taskRunner, interruptThread = interruptThread, reason = reason)
```

Comment:
showCreateReaper special cases thread interruption (and missing ofcourse) - specifying "reason" is trying to piggy back on top of this.
What is the developer expectation when killTask is invoked with a message - should the 'reason' override the reason propagated by the TaskReaper ? If it might not, we need to document this - the reason might not make it even if kill happens after api invocation.